### PR TITLE
Add order calendar

### PR DESCRIPTION
**Pull request by jdevega**

### CODE REVIEW

**Comment by jdevega** [comment]
I've removed the src/config.js file from source control ( added to .gitignore ), so you have a config.example.js you must use to create and customize your environment config on a new config.js file.

File: src/components/calendar/day.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
+// day.js
+
+import React, { Component, PropTypes } from 'react';
+import moment from 'moment';
+
+class Day extends Component {
+
```
Comment (kanedaki) [inline src/components/calendar/day.js]: Please delete all empty lines like there is in all other places

File: src/components/calendar/day.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    } else {
+      const mDate     = moment(date*1000);
+      value           = mDate.isValid() ? mDate.date() : '-';
+      dateValue       = mDate.isValid() ? mDate.valueOf() : null;
+    }
+    const tdStyle = this._getTdStyle();
+    const valueStyle = this._getDateValueStyle();
+
+    return(
+      <td data-date={dateValue} style={tdStyle} className={states.join(' ')} onClick={(event) => this.handleClick(event)} key={key}>
```
Comment (kanedaki) [inline src/components/calendar/day.js]: Why not call handleClick directly?

File: src/components/calendar/index.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  table: {
+    width:          '70%',
+    maxWidth:       '70%',
+    marginBottom:   20,
+    borderSpacing:  0,
+    borderCollapse: 'collapse'
+  }
+}
+
+export default Calendar;
```
Comment (kanedaki) [inline src/components/calendar/index.js]: why not do the export on the definition?

File: src/components/list-orders.js [hunk 4]
```diff
@@ -24,6 +25,7 @@ class ListOrders extends Component {
             </li>)
           }
         </ul>
+        {!isFetching && <CalendarOrder />}
```
Comment (kanedaki) [inline src/components/list-orders.js]: Do render the calendar always. The loading will be an overlay if isFetching is true.

File: src/index.js [hunk 5]
```diff
@@ -37,3 +37,7 @@ render(
   </I18nextProvider>,
   document.getElementById('root')
 )
+
+
+import moment from 'moment';
+window.moment = moment;
```
Comment (kanedaki) [inline src/index.js]: why do we need moment on the global namespace?

File: src/modules/orders/calendar.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+      })
+      break;
+    case actions.CALENDAR_SET_INITIAL_RANGE_DATE:
+      return Object.assign({}, state, {
+        initialRangeDate: action.date
+      })
+      break;
+    case actions.CALENDAR_ADD_SELECTED_DAY:
+      return Object.assign({}, state, {
+        selected: state.selected.concat(action.date)
```
Comment (kanedaki) [inline src/modules/orders/calendar.js]: to late to be sure but this is a mutation, right?

File: src/smart/calendar-order.js [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+  },
+  selected: {
+    backgroundColor: 'rgba(34, 40, 224, 0.63)',
+    color: 'white',
+    fontWeight: 'bold'
+  }
+} 
+
+function mapStateToProps(state, ownProps) {
+  const { orders, calendar } = state;
```
Comment (kanedaki) [inline src/smart/calendar-order.js]: orders should be fetched from the componentdidmount

Comment (kanedaki) [inline src/smart/calendar-order.js]: some test would be awesome. Please follow Julian guideline shown in base-app

File: src/index.js (hunk 5 above)
Comment (jdevega) [inline src/index.js]: Just for testing , removed !

**Pull request merged**